src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx

```diff
@@ -0,0 +1,87 @@
+import {getAxiosErrorMetadata} from '@atb/api/utils';
+import {Confirm} from '@atb/assets/svg/mono-icons/actions';
+import {
+  BottomSheetContainer,
+  useBottomSheet,
+} from '@atb/components/bottom-sheet';
+import {Button} from '@atb/components/button';
+import {MessageInfoBox} from '@atb/components/message-info-box';
+import {GenericSectionItem, Section} from '@atb/components/sections';
+import {ThemeText} from '@atb/components/text';
+import {StyleSheet} from '@atb/theme';
+import {consumeCarnet} from '@atb/ticketing';
+import {FareContractTexts, useTranslation} from '@atb/translations';
+import React, {useState} from 'react';
+import {ScrollView} from 'react-native-gesture-handler';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+type Props = {
+  fareContractId: string;
+};
+
+export const ConsumeCarnetBottomSheet = ({fareContractId}: Props) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<boolean>(false);
+  const {close} = useBottomSheet();
+
+  const onConsume = async () => {
+    setIsLoading(true);
+    try {
+      await consumeCarnet(fareContractId);
+      close();
+    } catch (e: any) {
+      const errorData = getAxiosErrorMetadata(e);
+      console.error('Error when consuming carnet: ', errorData);
```

Comment:
Noticing we should probably handle errors better than this. I guess a bugsnag report would make sense here.